openedx/core/djangoapps/content/block_structure/block_structure.py

```diff
@@ -464,7 +467,20 @@ def get_xblock_field(self, usage_key, field_name, default=None):
                 not found.
         """
         block_data = self._block_data_map.get(usage_key)
-        return getattr(block_data, field_name, default) if block_data else default
+        xblock_field = getattr(block_data, field_name, default) if block_data else default
+
+        if isinstance(xblock_field, datetime):
+            # Creating a new datetime object to avoid issues occuring due to upgrading
+            # python-datetuil version from 2.4.0
+            #
+            # More info: https://openedx.atlassian.net/browse/BOM-2245
+            if isinstance(xblock_field.tzinfo, tzlocal) and not hasattr(xblock_field.tzinfo, '_hasdst'):
+                return datetime(
+                    year=xblock_field.year, month=xblock_field.month, day=xblock_field.day,
+                    hour=xblock_field.hour, minute=xblock_field.minute, second=xblock_field.second,
+                    tzinfo=tzlocal()
```

Comment:
If we check the course_outline on sandbox where `USE_TZ = True`, we will find that's `tzlocal` is being used multiple times


Code snippet to check the course outline:
```python
from django.contrib.auth.models import User
from django.test import RequestFactory
from openedx.features.course_experience.utils import get_course_outline_block_tree

user = User.objects.get(username='admin')
request = RequestFactory().get(u'/')
request.user = user
course_key_string = 'course-v1:edX+DemoX+Demo_Course'


course_outline = get_course_outline_block_tree(request, course_key_string, request.user)
print(course_outline)
```

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/42294172/169003414-469f898f-d10d-4a3a-9ac6-f7f6f9aa3593.png">

Reply:
Ah, I forgot that this is coming from a cache instead of the database.  Yes, in that sense it makes sense to just replicate the time zone they had before being cached.